### ISSUE TITLE
How to train independent multi-classes?

### ISSUE THREAD

**Issue by topoincare** [qubvel/segmentation_models#308]
Hi, I have a problem about how to train independent classes. That mean, I have many classes: Road, Line, traffic signs (Left,Right,Straight). 
![goodgame_raw_965](https://user-images.githubusercontent.com/44630913/75806040-87fa1a00-5db5-11ea-9025-7ac574fd2d24.png)
And this mask
![goodgame_raw_965_L](https://user-images.githubusercontent.com/44630913/75806080-96e0cc80-5db5-11ea-940f-76488a5ea3c3.png)
But, I want to classify and segment traffic sign with label one class in image. That looks like:
![goodgame_bien_bao_raw_3950](https://user-images.githubusercontent.com/44630913/75806333-0787e900-5db6-11ea-9dda-a6528e08fd27.jpg)
![goodgame_bien_bao_raw_3950](https://user-images.githubusercontent.com/44630913/75806339-09ea4300-5db6-11ea-9fea-92fdfadcc10d.png)
If I label all classes in image. This makes me take long time. I want to label fast, so I label lots of frames have only traffic sign. My goal is predict all classes in image. 
How to train, which loss function and metrics I need? 
Thank you very much!

**Comment by JordanMakesMaps** [comment]
There isn't a way to get around labeling your data unless you have the money to pay someone else to do it for you. For the traffic signs (and maybe even the roads), you might be able to re-purpose real world datasets for training self-driving cars, and only keep the labels that correspond to traffic signs and road. 

Something else you can try is training a segmentation model on one of those datasets, then once fully trained, use it to label your own private dataset and keep the labels to train another segmentation model.

If you have multiple classes you should start with using one of the categorical loss functions, maybe start with `categorical cross-entropy` and included weights for your classes since they are going to be unbalanced. 

Metrics depends on you, what's most important. But IOU, Dice, Precision and Recall are all good, Pixel Accuracy is the only one I tell people to not put so much emphasis on because it's misleading.

Training is as easy as following the steps on the repo's readme page and also the notebooks in the examples folder.
